casadm/cas_lib.c

```diff
@@ -2691,9 +2694,10 @@ int list_caches(unsigned int list_format)
 			"-" /* device */);
 		for (i = 0; i < core_pool_path_cmd.core_pool_count; i++) {
 			char *core_path = core_pool_path_cmd.core_path_tab + (MAX_STR_LEN * i);
-			if (get_dev_path(core_path, core_path, MAX_STR_LEN)) {
-				cas_printf(LOG_WARNING, "WARNING: Can not resolve path to core. "
-						"By-id path will be shown for that core.\n");
+			if (!full_path){
+				if (get_dev_path(core_path, core_path, MAX_STR_LEN)) {
+					return FAILURE;
```

Comment:
You have files opened in :2659 and thread started in :2670. In case of error, you have to handle them, or return to original solution - print warning and the path in the way you can.